Libraries/LibWeb/CSS/StyleValues/CSSLightDark.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2025, Ladybird contributors
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "CSSLightDark.h"
+#include <LibWeb/Layout/Node.h>
+
+namespace Web::CSS {
+
+Color CSSLightDark::to_color(Optional<Layout::NodeWithStyle const&> node) const
+{
+    if (node.has_value() && node.value().computed_values().color_scheme() == PreferredColorScheme::Dark)
+        return m_properties.dark->to_color(node);
+
+    return m_properties.light->to_color(node);
+}
+
+bool CSSLightDark::equals(CSSStyleValue const& other) const
+{
+    if (type() != other.type())
+        return false;
+    auto const& other_color = other.as_color();
+    if (color_type() != other_color.color_type())
+        return false;
+    auto const& other_light_dark = verify_cast<CSSLightDark>(other_color);
+    return m_properties == other_light_dark.m_properties;
+}
+
+String CSSLightDark::to_string(SerializationMode mode) const
+{
+    // FIXME: We don't have enough information to determine the computed value here.
```

Comment:
Is this an actual problem, or has the `light-dark()` function been replaced with an actual color value by then? (Genuine question; this stuff is very confusing. :sweat_smile:)

Reply:
I don't think the `light-dark` style value is actually "replaced" at any point, this is relevant to my point in the PR description that our style computation system can't reasonably handle this.

Reply:
I just tried this PR out with http://wpt.live/css/css-color/light-dark-inheritance.html and we currently fail it. From that it seems we're supposed to resolve `light-dark()` before inheriting it. That's related to what I mean: the computed value can't be `light-dark()` itself, but should be whatever `light-dark()` has been resolved to.

Reply:
Yes, that's what should happen, unfortunately (as I have said) I am unable to find any way to make the style computer actually compute the style value. This is the same problem we had with `PositionStyleValue` which was "solved" with the `SerializationMode` workaround, that simply isn't enough here.

Reply:
Gotcha, I was a bit confused about what you meant exactly. That's a tricky one. :thinking:

Reply:
Nah, it's good as-is!